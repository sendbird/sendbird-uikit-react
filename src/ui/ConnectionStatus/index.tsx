import React, { ReactElement, useContext } from 'react';

import Icon, { IconTypes, IconColors } from '../Icon';
import Label, { LabelTypography, LabelColors } from '../Label';
import { LocalizationContext } from '../../lib/LocalizationContext';

import './index.scss';

function ConnectionStatus(): ReactElement {
  const { stringSet } = useContext(LocalizationContext);
  return (
    <div className="sendbird-connection-status">
      <Label
        type={LabelTypography.BODY_2}
        color={LabelColors.ONBACKGROUND_2}
      >
        {stringSet.TRYING_TO_CONNECT}
      </Label>
      <Icon
        type={IconTypes.DISCONNECTED}
        fillColor={IconColors.SENT}
        width="14px"
        height="14px"
      />
    </div>
  );
}

export default ConnectionStatus;
