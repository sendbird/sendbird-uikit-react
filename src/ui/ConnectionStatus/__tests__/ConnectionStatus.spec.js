import React from 'react';
import { render } from '@testing-library/react';

import ConnectionStatus from "../index";

describe('ui/ConnectionStatus', () => {
  it('should render ConnectionStatus', function () {
    const { asFragment } = render(
      <ConnectionStatus />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
