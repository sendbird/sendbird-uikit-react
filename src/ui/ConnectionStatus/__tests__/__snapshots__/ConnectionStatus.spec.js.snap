// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatus should render ConnectionStatus 1`] = `
<div
  className="sendbird-connection-status"
>
  <span
    className="sendbird-label sendbird-label--body-2 sendbird-label--color-onbackground-2"
  >
    Trying to connectâ€¦
  </span>
  <div
    className=" sendbird-icon sendbird-icon-disconnected sendbird-icon-color--sent"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    style={
      Object {
        "height": "14px",
        "minHeight": "14px",
        "minWidth": "14px",
        "width": "14px",
      }
    }
    tabIndex="0"
  >
    <test-file-stub />
  </div>
</div>
`;
